src/util.rs

```diff
@@ -199,30 +198,40 @@ macro_rules! map_to_owned (
 
 struct PrintState {
```

Comment:
Since has_predecessor no longer exists we can get rid of this struct

Reply:
I guess we can, i already wondered why that was a struct anyway